[torchgen] Rename executorch's RuntimeContext to KernelRuntimeContext

dbort opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #104892

Rename the context type to match changes in executorch.

Differential Revision: [D46977359](https://our.internmc.facebook.com/intern/diff/D46977359/)

dbort committed:
[torchgen] Rename executorch's RuntimeContext to KernelRuntimeContext

Rename the context type to match changes in executorch.

Differential Revision: [D46977359](https://our.internmc.facebook.com/intern/diff/D46977359/)

[ghstack-poisoned]

dbort committed:
Update on "[torchgen] Rename executorch's RuntimeContext to KernelRuntimeContext"


Rename the context type to match changes in executorch.

Differential Revision: [D46977359](https://our.internmc.facebook.com/intern/diff/D46977359/)

[ghstack-poisoned]

larryliu0820 commented on test/edge/kernel_runtime_context.h:
```diff
@@ -1,3 +1,5 @@
+// Copyright (c) Meta Platforms, Inc. and affiliates.
+
```
I think all files in pytorch/pytorch are covered by some license and we don't need this header

dbort replied:
Ah I was just copying the latest version of the internal header to here, and this came along. I can remove it.

dbort committed:
Update on "[torchgen] Rename executorch's RuntimeContext to KernelRuntimeContext"


Rename the context type to match changes in executorch.

Differential Revision: [D46977359](https://our.internmc.facebook.com/intern/diff/D46977359/)

[ghstack-poisoned]

dbort committed:
Update on "[torchgen] Rename executorch's RuntimeContext to KernelRuntimeContext"


Rename the context type to match changes in executorch.

Differential Revision: [D46977359](https://our.internmc.facebook.com/intern/diff/D46977359/)

[ghstack-poisoned]

dbort committed:
Update on "[torchgen] Rename executorch's RuntimeContext to KernelRuntimeContext"


Rename the context type to match changes in executorch.

Differential Revision: [D46977359](https://our.internmc.facebook.com/intern/diff/D46977359/)

[ghstack-poisoned]

dbort commented:
@pytorchbot merge
